### PR TITLE
[networks] Remove reliance on external IPs during offset guessing

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Ensure that the system-probe doesn't need to reach `8.8.4.4:53` to guess UDP offsets.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

File: pkg/network/tracer/offsetguess.go [hunk 1]
```diff
@@ -718,14 +718,20 @@ func newEventGenerator(ipv6 bool) (*eventGenerator, error) {
 
 	go acceptHandler(l)
 
+	// Spin up UDP server
+	udpAddr, udpDone, err := newUDPServer(addr)
```
Comment (brycekahle) [inline pkg/network/tracer/offsetguess.go:722]: Is spinning up a server necessary? We don't actually need the server to respond.

Comment (p-lambert) [reply pkg/network/tracer/offsetguess.go:722]: So you're suggesting we get rid of `net.Dial` altogether?

Comment (brycekahle) [reply pkg/network/tracer/offsetguess.go:722]: Can we use `127.0.0.2` (not 127.0.0.1 to distinguish from the source) without spinning up a server?

Comment (p-lambert) [reply pkg/network/tracer/offsetguess.go:722]: Oh, that might actually work. Let me try

Comment (p-lambert) [reply pkg/network/tracer/offsetguess.go:722]: Yeah, that doesn't work :(
The `net.Dial` function in Golang actually does a `connect` syscall (even for UDP), and apparently that fails if you don't have an actual server on the other side. I'm not sure if there some ICMP magic going on to check that, but I guess an workaround would actually create a UDP socket ourselves and do something like a `sendto`, but I'm guessing this is more work than the code I have.

Comment (hmahmood) [reply pkg/network/tracer/offsetguess.go:722]: Does that only fail if the address is loopback?

**Review by hmahmood — APPROVED** [review]

**Pull request merged**